src/inventorymanager.cpp

```diff
@@ -531,6 +531,17 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 		}
 	}
 
+	if (from_inv.type == InventoryLocation::PLAYER || to_inv.type == InventoryLocation::PLAYER){
+		// if it's moved in the selected slot and there are no items like such already
+		if (to_i == player->getWieldIndex() &&
+			to_stack_was.name != src_item.name)
+			PLAYER_TO_SA(player)->item_OnWield(src_item, player);
+		// if the whole stack is moved away from the selected slot
+		else if (from_i == player->getWieldIndex() &&
+				 count == from_stack_was.count)
+			PLAYER_TO_SA(player)->item_OnUnwield(src_item, player);
+	}
```

Comment:
I also don't see any handling for swaps here, it needs to call <i>both</i> unwield on the old item and wield on the new one